### PR TITLE
refactor(run_out): add state machine class for state transition 

### PR DESCRIPTION
**Pull request by TomohitoAndo**
## Description

- Refactored state transition
- Added documents.

NOTE: This PR doesn't change the actual behavior because this is just refactoring.

changes for tier4/autoware_launch
https://github.com/tier4/autoware_launch/pull/495

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: launch/tier4_planning_launch/config/scenario_planning/lane_driving/behavior_planning/behavior_velocity_planner/run_out.param.yaml [hunk 1]
```diff
@@ -30,9 +30,12 @@
...
         limit_vel_kmph: 3.0   # [km/h] limit velocity for approaching after stopping
-        stop_thresh: 0.01     # [m/s] threshold to decide if ego is stopping
-        stop_time_thresh: 3.0 # [sec] threshold for stopping time to transit to approaching state
-        dist_thresh: 0.5      # [m] end the approaching state if distance to the obstacle is longer than stop_margin + dist_thresh
+
+      # parameters for the change of state. used only when approaching is enabled
+      state:
+        stop_thresh: 0.01          # [m/s] threshold to decide if ego is stopping
+        stop_time_thresh: 3.0      # [sec] threshold for stopping time to transit to approaching state
+        disable_approach_dist: 4.0 # [m] end the approaching state if distance to the obstacle is longer than this value
```
Comment (0x126) [inline launch/tier4_planning_launch/config/scenario_planning/lane_driving/behavior_planning/behavior_velocity_planner/run_out.param.yaml:38]: Is it ok that `disable_approach_dist` has a different value than before?
Before this PR, stop_margin + dist_thresh = 2.5 + 0.5 = 3.0

Comment (TomohitoAndo) [reply launch/tier4_planning_launch/config/scenario_planning/lane_driving/behavior_planning/behavior_velocity_planner/run_out.param.yaml]: Thank you, it was a mistake.
fixed https://github.com/autowarefoundation/autoware.universe/pull/1884/commits/659e4d6f6c4e6d28c1cbc6c5b062b8d2c43becd5

**Review by 0x126 — APPROVED** [review]
LGTM

**Comment by TomohitoAndo** [comment]
@TakaHoribe 
If there are no problems, could you approve this as code owner of tier4_planning_lancuh package?

**Comment by kosuke55** [comment]
(FYI: I added other owners of `tier4_planning_lancuh` https://github.com/autowarefoundation/autoware.universe/pull/1955)

**Comment by TomohitoAndo** [comment]
@tkimura4 @shmpwk 
If there are no problems, could you approve this as maintainer of the package?

**Review by shmpwk — APPROVED** [review]
LGTM! maybe the config should be reflected to https://github.com/tier4/autoware_launch/pulls also?

**Comment by TomohitoAndo** [comment]
@shmpwk 
> maybe the config should be reflected to https://github.com/tier4/autoware_launch/pulls also?
yes! I will create the PR for tier4/autoware_launch and post a link in this PR.

**Comment by TomohitoAndo** [comment]
I created PR for tier4/autoware_launch
- https://github.com/tier4/autoware_launch/pull/495

**Pull request merged**